src/geode/geosciences_io/mesh/fem_output.cpp

```diff
@@ -0,0 +1,383 @@
+        }
+
+        void write_one_coord_component_dist(
+            const std::string dim_name, const int dim )
+        {
+            file_ << dim_name << EOL;
+            absl::flat_hash_map< double, std::vector< int > > dist =
+                create_coord_dist( dim );
+            std::vector< double > values;
+            values.reserve( dist.size() );
+            for( auto val = dist.begin(); val != dist.end(); ++val )
+            {
+                values.push_back( val->first );
+            }
+            std::sort( values.begin(), values.end() );
+            for( const auto v : values )
+            {
+                std::string line = "";
+                for( const auto vtx : dist[v] )
+                {
+                    line += std::to_string( vtx + 1 ) + SPACE;
+                }
+                file_ << "     " << v << "  " << line << EOL;
+            }
+        }
+
+        void write_xyz_dist()
+        {
+            write_one_coord_component_dist( "X", 0 );
+            write_one_coord_component_dist( "Y", 1 );
+            write_one_coord_component_dist( "Z", 2 );
+        }
+        // void write_node_property( geode::AttributeBase& attribute )
+        // {
+        //     file_ << attribute.name() << EOL;
+        //     absl::flat_hash_map< double, std::vector< int > > att_dist =
+        //         create_att_dist( attribute );
+        //     // const auto value = attribute.generic_value( elem );
+        //     std::vector< double > values;
+        //     values.reserve( att_dist.size() );
+        //     for( auto val = att_dist.begin(); val != att_dist.end(); ++val )
+        //     {
+        //         values.push_back( val->first );
+        //     }
+        //     std::sort( values.begin(), values.end() );
+        //     for( const auto v : values )
+        //     {
+        //         std::string line = "";
+        //         for( const auto e : att_dist[v] )
+        //         {
+        //             line += std::to_string( e + 1 ) + SPACE;
+        //         }
+        //         file_ << "     " << v << "  " << line << EOL;
+        //     }
+        // }
+
+        void write_ref_element_dist()
+        {
+            file_ << "REF_DISE_I" << EOL;
+            int nb_elem_att =
+                solid_.polyhedron_attribute_manager().attribute_names().size()
+                - 3;
+            file_ << SPACE << nb_elem_att << SPACE << solid_.nb_polyhedra()
+                  << " 0" << EOL;
+            for( const auto name :
+                solid_.polyhedron_attribute_manager().attribute_names() )
+            {
+                const auto attribute = solid_.polyhedron_attribute_manager()
+                                           .find_generic_attribute( name );
+                if( !attribute || !attribute->is_genericable()
+                    || name == "tetrahedron_vertices"
+                    || name == "tetrahedron_adjacents"
+                    || name == "geode_active" )
```

Comment:
why removing them?

Reply:
I did'nt find these attributes in fem files